File: 프로그래머스/lv1/72410. 신규 아이디 추천/신규 아이디 추천.java
import java.util.*;
class Solution {
    public String solution(String new_id) {
        String answer = "";
        
        new_id=new_id.toLowerCase();
        new_id=new_id.replaceAll("[^0-9a-z\\.\\-\\_]","");
        new_id=new_id.replaceAll("\\.{2,}",".");
        if(new_id.charAt(0)=='.')   new_id=new_id.substring(1,new_id.length());
        if(new_id.length()>0&&new_id.charAt(new_id.length()-1)=='.')   new_id=new_id.substring(0,new_id.length()-1);
        if(new_id.length()==0)  new_id="a";
        else if(new_id.length()>15) new_id=new_id.substring(0,15);
        
        if(new_id.length()>0&&new_id.charAt(new_id.length()-1)=='.')   new_id=new_id.substring(0,new_id.length()-1);
        if(new_id.length()<=2){
            String str=new_id.substring(new_id.length()-1,new_id.length());
            for(int i=0;i<=3-new_id.length();i++)
                new_id+=str;
        }
        answer=new_id;
        return answer;
    }
}

File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.py
def solution(new_id):
    answer = ''
    
    new_id=new_id.lower()
    for w in new_id:
        if w.isalnum() or w in '-_.':
            answer+=w
            
    while '..' in answer:
        answer=answer.replace('..','.')

    if len(answer)>0 and answer[0]=='.':
        answer=answer[1:]
    if len(answer)>0 and answer[-1]=='.':
        answer=answer[:-1]
    
    if len(answer)==0:
        answer='a'
    if len(answer)>=16:
        answer=answer[:15]
        if answer[-1]=='.':
            answer=answer[:-1]
    
    while len(answer)<3:
        answer+=answer[-1]

    return answer